File: Excel_Programs/C++/question5.cpp
// Move all the negative elements to one side of the array 
#include<iostream>
using namespace std;

int main()
{
    int arr[] = {-12, 11, -13, -5, 6, -7, 5, -3, -6};
    int neg=0,pos=0;
    for(int i=0;i<(sizeof(arr)/4);i++)
    {
        if(arr[i]<0)
        {
            neg++;
        }
        if(arr[i]>0)
        {
            pos++;
        }
    }
    int arrneg[neg],arrpos[pos];
    for(int i =0,j=0,k=0;i<sizeof(arr)/4;i++)
    {
        if(arr[i]<0)
        {
            arrneg[j]=arr[i];
            j++;
        }
        if(arr[i]>0)
        {
            arrpos[k]=arr[i];
            //cout<<arr[i]<<"\n";
            k++;
        }
    }
    int x=0;
    for(int i=0,j=0;i<neg;i++)
    {
        arr[i]=arrneg[i];
    }
    for(int i=0;i<pos;i++)
    {
        arr[i+neg] = arrpos[i];
    }
    for(int i=0;i<sizeof(arr)/4;i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<"\n";
    return 100;
}

File: Excel_Programs/Java/question5.java
// Move all the negative elements to one side of the array 
import java.util.*;
class question5 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of elements to be entered");
        int n = sc.nextInt();
        int arr[] = new int[n];
        for (int i =0;i<n;i++)
        {
            System.out.print("Enter a number : ");
            arr[i] = sc.nextInt();
        }
        int neg=0,pos=0;
        for(int i=0;i<arr.length;i++)
        {
            if(arr[i]<0)
                neg++;
            if(arr[i]>0)
                pos++;
        }
        int arrneg[] = new int[neg];
        int arrpos[] = new int[pos];

        for(int i = 0,j=0,k=0;i<arr.length;i++)
        {
            if(arr[i]<0)
            {
                arrneg[j]=arr[i];
                j++;
            }
            if(arr[i]>0)
            {
                arrpos[k]=arr[i];
                k++;
            }
        }
        for(int i=0 ;i<neg;i++)
        {
            arr[i] =arrneg[i];
        }
        for(int i=0;i<pos;i++)
        {
            arr[neg+i]=arrpos[i];
        }
        for(int i=0;i<arr.length;i++)
        {
            System.out.print(arr[i]+" ");
        }
        System.out.println();

        sc.close();

    }
}
